app/code/core/Mage/Paypal/Model/Hostedpro/Request.php

```diff
@@ -157,7 +157,7 @@ protected function _getOrderData(Mage_Sales_Model_Order $order)
     {
         $request = array(
             'subtotal'      => $this->_formatPrice($order->getBaseSubtotal()),
-            'tax'           => $this->_formatPrice($order->getBaseTaxAmount()),
+            'tax'           => $this->_formatPrice($order->getBaseTaxAmount() + $order->getHiddenTaxAmount() ),
```

Comment:
customer places an order with discount which generates hidden tax in magento. PayPal does its own discount and tax calculation, without the mod paypal marks the order as fraud as the magneto and paypal totals dont match. Others have found the same problem

https://magento.stackexchange.com/questions/49464/magento-1-9-1-bug-discount-tax-paypal-pro-incorrect-total